core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -38,7 +38,14 @@
 .section .data
 .balign 4
 
-#ifdef CFG_BOOT_SYNC_CPU
+/*
+ * For VxWorks Support, the primary core will go through OP-TEE init
+ * without ever touching secondary cores; it is only after the primary
+ * core enters VxWorks, it issues special SMC to request booting of
+ * secondary cores. Thus CPU Sync is not required.
+ */
+
+#if defined (CFG_BOOT_SYNC_CPU) && !defined (CFG_TEE_VXWORKS_SUPPORT)
```

Comment:
I understand we can set CFG_BOOT_SYNC_CPU=n in plat-xxx/conf.mk. However, the intention here is to "uniformly" disable the CPU sync feature if OP-TEE is to be used together with VxWorks. Besides, for some platform, for example, plat-ls, it already defines the CFG_BOOT_SYNC_CPU=y. We want to reuse the existing plat-ls to support VxWorks, but we cannot just rudely overwrite CFG_BOOT_SYNC_CPU=n, so we have to disable the feature using the CFG_TEE_VXWORKS_SUPPORT flag. 

It seems the generic boot process implementation is targeted for Firmware Code. The firmware would bring up all cores and put each through the OP-TEE; VxWorks's usage differs from this: only the primary core is required to go through the OP-TEE and enters NS OS (this maximize compatibility as it is not required to have TrustZone aware firmware), then NS OS asks the OP-TEE to bring up the rest cores, so there's no need to sync between primary and secondary cores.